src/System.Private.CoreLib/shared/System/Enum.cs

```diff
@@ -906,6 +906,57 @@ public override string ToString()
             // Try to see if its one of the enum values, then we return a String back else the value
             return InternalFormat((RuntimeType)GetType(), ToUInt64()) ?? ValueToString();
         }
+
+        public int CompareTo(object? target)
+        {
+            if (target == this)
+                return 0;
+
+            if (target == null)
+                return 1; // all values are greater than null
+
+            if (this.GetType() != target.GetType())
```

Comment:
Although I prefer the `this.` prefix, I thought this was supposed to be avoided in product code according to (4) in the [coding guidelines](https://github.com/dotnet/corefx/blob/dc71f95667cabd5ad0f7a051c974e0d312b243e8/Documentation/coding-guidelines/coding-style.md#c-coding-style).